raiden/storage/serialization/cache.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any, Dict
+
+from marshmallow import Schema
+from marshmallow_dataclass import class_schema
+
+
+def bind(instance, func):
+    """
+    Bind the function *func* to *instance*.
+    The provided *func* should accept the
+    instance as the first argument, i.e. "self".
+    """
+    bound_method = func.__get__(instance, instance.__class__)
+    setattr(instance, func.__name__, bound_method)
+    return bound_method
+
+
+def class_type(instance: Any) -> str:
+    return f"{instance.__class__.__module__}.{instance.__class__.__name__}"
+
+
+def set_class_type(schema, data, instance):  # pylint: disable=unused-argument
+    data["_type"] = class_type(instance)
```

Comment:
This is basically adding the type on the top-level class that we need to know when deserializing... previously, type was attached to every class included nested ones. 
If i wanted to use a mapping, we would need to answer those questions:
1. How would you know which class the JSON dict represents?
2. How are we going to populate the mapping with all of our classes on restarts?

Reply:
> How would you know which class the JSON dict represents?

We still need the `_type` attribute.

> How are we going to populate the mapping with all of our classes on restarts?

A mapping like the one we have in [_native_to_marshmallow.update](https://github.com/raiden-network/raiden/pull/3969/files/05ad06d8f7f1d67e01c2cbe45947f09c0bc72262#diff-86c4c100c6f176a3ea18c6a69a790af8)

Reply:
Note: BTW, we can totally do this once that is necessary, I'm just pointing out that having a hard coded path to the class is not sufficient. (Maybe this could be fixed with migrations)